### PR TITLE
[GOBBLIN-1496] Initialize arrays and maps related to gobblin as a service with an in…

### PR DESCRIPTION
**Pull request by Will-Lo**
…itial size if applicable

Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1496

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):
GaaS has many scenarios with sparse/empty maps and lists. When initializing new data structures, we should initialize them to a certain size if we know the elements beforehand.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by Will-Lo** [comment]
@jack-moseley @arjun4084346 @umustafi could you review?

File: gobblin-modules/gobblin-http/src/main/java/org/apache/gobblin/utils/HttpUtils.java [hunk 1]
```diff
@@ -143,7 +143,7 @@ public static URI buildURI(String urlTemplate, Map<String, String> keys, Map<Str
    */
   public static void updateStatusType(ResponseStatus status, int statusCode, Set<String> errorCodeWhitelist) {
     if (statusCode >= 300 & statusCode < 500) {
-      List<String> whitelist = new ArrayList<>();
```
Comment (umustafi) [inline gobblin-modules/gobblin-http/src/main/java/org/apache/gobblin/utils/HttpUtils.java:146]: We should try changing this name to "allow/denyList" as a part of this initiative for more inclusive language: https://iwww.corp.linkedin.com/wiki/cf/pages/viewpage.action?pageId=372446381

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/flow/FlowGraphPath.java [hunk 2]
```diff
@@ -142,26 +142,26 @@ private static boolean isNodeForkable(DagNode<JobExecutionPlan> dagNode) {
...
-    //Iterate over each resolved job config and convert the config to a JobSpec.
-    for (Config resolvedJobConfig : resolvedJobConfigs) {
-      JobExecutionPlan jobExecutionPlan = new JobExecutionPlan.Factory().createPlan(flowSpec, resolvedJobConfig, specExecutor, flowExecutionId, sysConfig);
-      jobExecutionPlans.add(jobExecutionPlan);
-      templateToJobNameMap.put(getJobTemplateName(jobExecutionPlan), jobExecutionPlan.getJobSpec().getConfig().getString(
-          ConfigurationKeys.JOB_NAME_KEY));
-    }
-    updateJobDependencies(jobExecutionPlans, templateToJobNameMap);
-    return new JobExecutionPlanDagFactory().createDag(jobExecutionPlans);
+     FlowTemplate flowTemplate = flowEdgeContext.getEdge().getFlowTemplate();
```
Comment (umustafi) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/flow/FlowGraphPath.java:145]: Was the formatting off for this original code block?

**Review by jack-moseley — APPROVED** [review]

**Review by phet — COMMENTED** [review]
`>=` - shouldn't "400 Bad Request" fall within `CODE_4XX`?

File: gobblin-modules/gobblin-http/src/main/java/org/apache/gobblin/utils/HttpUtils.java (hunk 1 above)
Comment (Will-Lo) [reply gobblin-modules/gobblin-http/src/main/java/org/apache/gobblin/utils/HttpUtils.java:146]: We can have a new PR to include all of it throughout the codebase, good call out!

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/flow/FlowGraphPath.java (hunk 2 above)
Comment (Will-Lo) [reply gobblin-service/src/main/java/org/apache/gobblin/service/modules/flow/FlowGraphPath.java:145]: Yeah it was slightly off here, the autoformat fixed it when i modified the mile

**Comment by Will-Lo** [comment]
@phet you're right, I can add the verification change on another PR (want to keep this PR empty of logic changes to speed up/simplify testing for fliptop right now)

**Comment by sv2000** [comment]
+1.

**Review by sv2000 — APPROVED** [review]

**Pull request merged**